test/SILOptimizer/experimental-swift-based-closure-specialization/closure_specialize.sil

```diff
@@ -0,0 +1,941 @@
+  store %0 to %5a : $*Int                        // id: %6
+  %7 = alloc_box $<τ_0_0> { var τ_0_0 } <Int>, var, name "q"                   // users: %8, %11, %13
+  %7a = project_box %7 : $<τ_0_0> { var τ_0_0 } <Int>, 0
+  store %1 to %7a : $*Int                        // id: %8
+  // function_ref specgen.callee (Swift.Int, Swift.Int, Swift.Int) -> ()
+  %9 = function_ref @$s7specgen6calleeyySi_S2itF : $@convention(thin) (Int, Int, Int) -> () // user: %12
+  %10 = load %5a : $*Int                         // user: %12
+  %11 = load %7a : $*Int                         // user: %12
+  %12 = apply %9(%10, %11, %2) : $@convention(thin) (Int, Int, Int) -> ()
+  strong_release %7 : $<τ_0_0> { var τ_0_0 } <Int>
+  strong_release %5 : $<τ_0_0> { var τ_0_0 } <Int>
+  %15 = tuple ()                                  // user: %16
+  return %15 : $()                                // id: %16
+}
+
+//////////////////////////////////
+// Thin To Thick Function Tests //
+//////////////////////////////////
+
+// CHECK-LABEL: sil [noinline] @$s7specgen6calleeyySi_SitF : $@convention(thin) (Int, Int) -> () {
+// specgen.callee (Swift.Int, Swift.Int) -> ()
+sil [noinline] @$s7specgen6calleeyySi_SitF : $@convention(thin) (Int, Int) -> () {
+bb0(%0 : $Int, %1 : $Int):
+  %6 = tuple ()                                   // user: %7
+  return %6 : $()                                 // id: %7
+}
+
+// CHECK-LABEL: sil @$s7specgen11tttficalleryySiF : $@convention(thin) (Int) -> () {
+// CHECK: [[ID1:%[0-9]+]] = function_ref @$s7specgen13take_closure2yyySi_SitcF26$s7specgen6calleeyySi_SitFTf1c_n : $@convention(thin) () -> ()
+// CHECK: [[ID2:%[0-9]+]] = function_ref @$s7specgen12take_closureyyySi_SitcF26$s7specgen6calleeyySi_SitFTf1c_n : $@convention(thin) () -> ()
+// CHECK: apply [[ID2]]() : $@convention(thin) () -> ()
+// CHECK: apply [[ID1]]() : $@convention(thin) () -> ()
+sil @$s7specgen11tttficalleryySiF : $@convention(thin) (Int) -> () {
+bb0(%0 : $Int):
+  // function_ref specgen.take_closure ((Swift.Int, Swift.Int) -> ()) -> ()
+  %2 = function_ref @$s7specgen12take_closureyyySi_SitcF : $@convention(thin) (@owned @callee_owned (Int, Int) -> ()) -> () // user: %5
+  // function_ref specgen.(caller (Swift.Int) -> ()).(closure #1)
+  %3 = function_ref @$s7specgen6calleeyySi_SitF : $@convention(thin) (Int, Int) -> () // user: %4
+  %4 = thin_to_thick_function %3 : $@convention(thin) (Int, Int) -> () to $@callee_owned (Int, Int) -> () // user: %5
+  %5 = apply %2(%4) : $@convention(thin) (@owned @callee_owned (Int, Int) -> ()) -> ()
+  %6 = function_ref @$s7specgen13take_closure2yyySi_SitcF : $@convention(thin) (@owned @callee_owned (Int, Int) -> ()) -> ()
+  %7 = apply %6(%4) : $@convention(thin) (@owned @callee_owned (Int, Int) -> ()) -> ()
+  %9999 = tuple ()                                   // user: %7
+  return %9999 : $()                                 // id: %7
+}
+
+// We don't handle closures that close over address types (*NOTE* this includes
```

Comment:
TODO: ?

Reply:
There was a TODO in the ClosureSpecialization.swift. Added it here as well.